### PR TITLE
KSES: Improve compatibility for WP 5.4.1+

### LINKED ISSUE #7631: KSES: Improve compatibility for WP 5.4.1+
## Feature Description

By leveraging a new param that was added to the `wp_insert_post_data` filter in WordPRess 5.4.1, we can better target posts of the web-story post type with our custom KSES logic.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

By leveraging a new param that was added to the `wp_insert_post_data` filter in WordPRess 5.4.1, we can better target posts of the web-story post type with our custom KSES logic.

The existing behavior is still in place for WP < 5.4.1, but should be removed once we drop support for those versions.

Originally discussed at google/web-stories-wp#6077

## User-facing changes

N/A

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Related: #7551 
Fixes #7631

### CODE REVIEW

File: includes/KSES.php [hunk 1]
```diff
@@ -47,13 +48,24 @@ public function register() {
...
-			add_filter( 'wp_kses_allowed_html', [ $this, 'filter_kses_allowed_html' ], 10, 2 );
-			add_filter( 'content_save_pre', [ $this, 'filter_content_save_pre_before_kses' ], 0 );
-			add_filter( 'content_save_pre', [ $this, 'filter_content_save_pre_after_kses' ], 20 );
-			remove_filter( 'content_filtered_save_pre', 'wp_filter_post_kses' );
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			return;
+		}
+
+		// The third param of the below filter was only added in WP 5.4.1.
+		if ( version_compare( get_bloginfo( 'version' ), '5.4.1', '>=' ) ) {
```
Comment (spacedmonkey) [inline includes/KSES.php:58]: As we also have some problems with 5.3 and gutenberg. It is worth having the conversion around making 5.4.1 the min version?

File: includes/KSES.php [hunk 2]
```diff
@@ -67,6 +79,47 @@ public static function get_registration_action_priority() {
...
+
+		$unsanitized_postarr['post_content'] = $this->filter_content_save_pre_before_kses( $unsanitized_postarr['post_content'] );
+		$data['post_content']                = wp_filter_post_kses( $unsanitized_postarr['post_content'] );
+		$data['post_content']                = $this->filter_content_save_pre_after_kses( $data['post_content'] );
+
+		remove_filter( 'safe_style_css', [ $this, 'filter_safe_style_css' ] );
+		remove_filter( 'wp_kses_allowed_html', [ $this, 'filter_kses_allowed_html' ] );
+
+		return $data;
+	}
```
Comment (spacedmonkey) [inline includes/KSES.php]: Nice 😎

File: includes/KSES.php (hunk 1 above)
Comment (swissspidy) [reply includes/KSES.php:58]: I think so, yeah. Last time I checked, most users are on 5.6+ anyway, so data would back this up too.

Comment (swissspidy) [reply includes/KSES.php:58]: Created #7551, will discuss with product team

File: tests/phpunit/tests/REST_API/Autosaves_Controller.php [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 namespace Google\Web_Stories\Tests\REST_API;
 
+use Google\Web_Stories\Story_Post_Type;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/REST_API/Autosaves_Controller.php:20]: Unrelated change?

File: includes/KSES.php [hunk 4]
```diff
@@ -47,13 +50,24 @@ public function register() {
...
-			add_filter( 'wp_kses_allowed_html', [ $this, 'filter_kses_allowed_html' ], 10, 2 );
-			add_filter( 'content_save_pre', [ $this, 'filter_content_save_pre_before_kses' ], 0 );
-			add_filter( 'content_save_pre', [ $this, 'filter_content_save_pre_after_kses' ], 20 );
-			remove_filter( 'content_filtered_save_pre', 'wp_filter_post_kses' );
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			return;
+		}
+
+		// The third param of the below filter was only added in WP 5.4.1.
+		if ( version_compare( get_bloginfo( 'version' ), '5.4.1', '>=' ) ) {
```
Comment (spacedmonkey) [inline includes/KSES.php:58]: Should we have a unit test or something to make sure this works?

Comment (swissspidy) [reply includes/KSES.php:58]: I think this is covered by the existing unit tests which run against both 5.3 and latest.

File: tests/phpunit/tests/REST_API/Autosaves_Controller.php (hunk 3 above)
Comment (swissspidy) [reply tests/phpunit/tests/REST_API/Autosaves_Controller.php:20]: Tiny cleanup I happened to add when testing failing autosaves test. Kept it for readability reasons.

**Review by spacedmonkey — APPROVED** [review]
